File: Chapter06/spring-boot-2-taxi/spring-boot-2-taxi-booking-service/src/main/java/com/packtpub/springboot2taxiservice/controller/BaseController.java
package com.packtpub.springboot2taxiservice.controller;

import com.packtpub.springboot2taximodel.dto.response.ErrorDTO;
import com.packtpub.springboot2taxiservice.exception.TaxiBookingIdNotFoundException;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.ControllerAdvice;
import org.springframework.web.bind.annotation.ExceptionHandler;

@ControllerAdvice
public class BaseController {

    @ExceptionHandler(TaxiBookingIdNotFoundException.class)
    public ResponseEntity<ErrorDTO> handleTaxiBookingIdNotFoundException(TaxiBookingIdNotFoundException e) {
        return new ResponseEntity<ErrorDTO>(new ErrorDTO(e.getMessage(), HttpStatus.BAD_REQUEST.value()), HttpStatus.BAD_REQUEST);
    }
}

File: Chapter05/spring-boot-2-movie-rating/src/main/kotlin/com/packtpub/springboot2movierating/controller/BaseController.kt
/*
    Copyright (C) 2018  Shazin Sadakath
    This program is free software: you can redistribute it and/or modify
    it under the terms of the GNU General Public License as published by
    the Free Software Foundation, either version 3 of the License, or
    (at your option) any later version.
    This program is distributed in the hope that it will be useful,
    but WITHOUT ANY WARRANTY; without even the implied warranty of
    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
    GNU General Public License for more details.
    You should have received a copy of the GNU General Public License
    along with this program.  If not, see <https://www.gnu.org/licenses/>.
 */
package com.packtpub.springboot2movierating.controller

import com.packtpub.springboot2movierating.dto.ErrorDTO
import com.packtpub.springboot2movierating.exception.MovieNotFoundException
import org.springframework.http.HttpStatus
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.ControllerAdvice
import org.springframework.web.bind.annotation.ExceptionHandler

@ControllerAdvice
class BaseController {

    @ExceptionHandler(value = MovieNotFoundException::class)
    fun handleMovieNotFoundException (e : MovieNotFoundException) : ResponseEntity<ErrorDTO> {
        return ResponseEntity<ErrorDTO>(ErrorDTO(400, e.message), HttpStatus.BAD_REQUEST);
    }
}
